File: exercise-2022/simulation-2016/1.cpp
#include <iostream>
using namespace std;

#define ll long long

ll gcd(ll a, ll b)
{ return !b ? a : gcd(b, a%b); }

ll lcm(ll a, ll b)
{ return a / gcd(a, b) * b; }

void solve()
{
    int n;
    cin >> n;
    ll fenzi = 0, fenmu = 1;
    for (int i=0; i<n; i++)
    {
        ll tmpzi, tmpmu;
        scanf("%lld/%lld", &tmpzi, &tmpmu);
        ll l = lcm(fenmu, tmpmu);
        fenzi = l/fenmu*fenzi + l/tmpmu*tmpzi;
        fenmu = l;
        ll g = gcd(fenzi, fenmu);
        fenzi /= g, fenmu /= g;
    }
    ll zhengshu = fenzi / fenmu;
    ll fenshu = fenzi % fenmu;
    if (zhengshu) printf("%lld", zhengshu);
    if (fenshu)
    {
        if (zhengshu) putchar(' ');
        printf("%lld/%lld", fenshu, fenmu);
    }
    if (!zhengshu && !fenshu) puts("0");
}

int main()
{
    solve();
    return 0;
}

File: 2016/simulation/1.py
def dev(a, b):
    return int(a / b)
def mod(a, b):
    return a - dev(a, b) * b
def gcd(a, b):
    return a if b==0 else gcd(b, mod(a, b))
def lcm(a, b):
    return dev(a, gcd(a, b)) * b

fenzi = 0
fenmu = 1
n = int(input())
for i in input().split(' '):
    newzi, newmu = map(int, i.split('/'))
    l = lcm(fenmu, newmu)
    fenzi = fenzi*dev(l, fenmu) + newzi*dev(l, newmu)
    fenmu = l
    g = gcd(fenzi, fenmu)
    fenzi = dev(fenzi, g)
    fenmu = dev(fenmu, g)
zhengshu = dev(fenzi, fenmu)
fenshu = mod(fenzi, fenmu)
flag = False
if zhengshu != 0:
    print(zhengshu, end='')
    flag = True
if fenshu != 0:
    if flag:
        print(' ', end='')
    print(fenshu, fenmu, sep='/')
elif not flag:
    print(0)
